### PR TITLE
3579 feedback page self hostedlatestupgradesminor upgrade

### LINKED ISSUE #3579: [Feedback] Page: /self-hosted/latest/upgrades/minor-upgrade/
### Is it easy to find the information you need?

No

### Are the instructions clear?

Yes

### How could we improve the Timescale documentation site?

Hello,

The article presents no solution for the most common error message:

```
# psql $SOURCE -c "ALTER EXTENSION timescaledb UPDATE TO '2.16.1';"
ERROR:  extension "timescaledb" has no update path from version "2.11.0" to version "2.16.1"
# psql $SOURCE -c "ALTER EXTENSION timescaledb UPDATE;"
ERROR:  extension "timescaledb" has no update path from version "2.11.0" to version "2.17.1"
```

Thank you in advance

### PR DESCRIPTION
**Pull request by billy-the-fish**
Update the upgrade instructions so users no longer have the same issue as https://github.com/timescale/docs/issues/3579.

### CODE REVIEW

File: _partials/_migrate_self_postgres_implement_migration_path.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+   Follow the procedure in [Upgrade PostgreSQL][upgrade-pg]. The version of TimescaleDB installed
+   in your PostgreSQL deployment must be the same before and after the PostgreSQL upgrade.
+
+1. **If your migration path dictates it, upgrade TimescaleDB again**
+
+    ```sql
+   psql -X -d $SOURCE -c "ALTER EXTENSION timescaledb UPDATE TO '<version number>';"
+   ```
+
+1. **Check that you have upgraded to the correct version TimescaleDB**
```
Comment (graveland) [inline _partials/_migrate_self_postgres_implement_migration_path.md]:
```suggestion
1. **Check that you have upgraded to the correct version of TimescaleDB**
```

File: _partials/_plan_upgrade.md [hunk 2]
```diff
@@ -1,12 +1,10 @@
...
 Before you upgrade:
 
-*   Read [the release notes][relnotes] for the Timescale version you are
-    upgrading to.
-*   Check which PostgreSQL version you are currently running. You might need to
+*   Read [the release notes][relnotes] for the version of TimescaleDB that you are upgrading to.
+*   Check the version TimescaleDB you are currently running, and the version you want to upgrade to
+    against the [upgrade matrix][upgrade-pg]. You may need to up 
+
+* You might need to
```
Comment (graveland) [inline _partials/_plan_upgrade.md]: Duplicate "You might need to"

File: self-hosted/upgrades/major-upgrade.md [hunk 3]
```diff
@@ -1,141 +1,153 @@
...
+[Upgrade TimescaleDB to a minor version][upgrade-minor].
 
-<PlanUpgrade />
+## Prerequisites
 
-## Breaking changes
+- Install the PostgreSQL client tools on your migration machine. This includes `psql`, and `pg_dump`.
+- Read [the release notes][relnotes] for the version of TimescaleDB that you are upgrading to.
+- Backup your database. Although TimescaleDB and PostgreSQL upgrades are performed in-place, upgrading is an
+  intrusive operation. In case of disaster, [create a database backup][backup], and ensure that the backup
```
Comment (graveland) [inline self-hosted/upgrades/major-upgrade.md]: Maybe?

```suggestion
  intrusive operation. To protect against disasters, [create a database backup][backup], and ensure that the backup
```

File: self-hosted/upgrades/minor-upgrade.md [hunk 4]
```diff
@@ -1,66 +1,51 @@
...
 ---
 
 import PlanUpgrade from "versionContent/_partials/_plan_upgrade.mdx";
 import ConsiderCloud from "versionContent/_partials/_consider-cloud.mdx";
+import CheckVersions from "versionContent/_partials/_migrate_self_postgres_check_versions.mdx";
+import PlanMigrationPath from "versionContent/_partials/_migrate_self_postgres_plan_migration_path.mdx";
+import ImplementMigrationPath from "versionContent/_partials/_migrate_self_postgres_implement_migration_path.mdx";
 
-# Minor TimescaleDB upgrades
+# Upgrade TimescaleDB to a minor version
```
Comment (graveland) [inline self-hosted/upgrades/minor-upgrade.md]:
```suggestion
# Upgrade TimescaleDB to a new minor version
```

File: self-hosted/upgrades/minor-upgrade.md [hunk 5]
```diff
@@ -1,66 +1,51 @@
...
-This upgrade uses the PostgreSQL `ALTER EXTENSION` function to upgrade to the
-latest version of the TimescaleDB extension. TimescaleDB supports having
-different extension versions on different databases within the same PostgreSQL
-instance. This allows you to upgrade extensions independently on different
-databases. Run the `ALTER EXTENSION` function on each database to upgrade them
-individually.
+- Install the PostgreSQL client tools on your migration machine. This includes `psql`, and `pg_dump`.
+- Read [the release notes][relnotes] for the version of TimescaleDB that you are upgrading to.
+- Backup your database. Although TimescaleDB and PostgreSQL upgrades are performed in-place, upgrading is an 
+    intrusive operation. In case of disaster, [create a database backup][backup], and ensure that the backup
```
Comment (graveland) [inline self-hosted/upgrades/minor-upgrade.md]: (same as the other related comment)

File: self-hosted/upgrades/upgrade-pg.md [hunk 6]
```diff
@@ -1,100 +1,92 @@
...
-`pg_upgrade` allows you to retain the data files of your current PostgreSQL
-installation while binding the new PostgreSQL binary runtime to them. This is
-supported for PostgreSQL&nbsp;8.4 and higher.
+
+## Prerequisites
+
+- Install the PostgreSQL client tools on your migration machine. This includes `psql`, and `pg_dump`.
+- Read [the release notes][pg-relnotes] for the version of PostgreSQL that you are upgrading to.
+- Backup your database. Although TimescaleDB and PostgreSQL upgrades are performed in-place, upgrading is an
+  intrusive operation. In case of disaster, [create a database backup][backup], and ensure that the backup
```
Comment (graveland) [inline self-hosted/upgrades/upgrade-pg.md]: (same)

**Review by graveland — APPROVED** [review]
lgtm, added a few comments though, nothing major

File: _partials/_plan_upgrade.md [hunk 7]
```diff
@@ -1,12 +1,10 @@
...
-that you do not need to dump and restore your data. However, it is still
-important that you plan for your upgrade ahead of time.
-
 Before you upgrade:
 
-*   Read [the release notes][relnotes] for the Timescale version you are
-    upgrading to.
-*   Check which PostgreSQL version you are currently running. You might need to
+*   Read [the release notes][relnotes] for the version of TimescaleDB that you are upgrading to.
+*   Check the version TimescaleDB you are currently running, and the version you want to upgrade to
```
Comment (atovpeko) [inline _partials/_plan_upgrade.md]:
```suggestion
*   Check the version of TimescaleDB you are currently running, and the version you want to upgrade to
```

File: self-hosted/upgrades/downgrade.md [hunk 8]
```diff
@@ -72,14 +85,19 @@ upgrading and downgrading.
     For example:
 
     ```sql
-    ALTER EXTENSION timescaledb UPDATE TO '2.5.1';
+    ALTER EXTENSION timescaledb UPDATE TO '2.17.0';
     ```
 
-1.  Check that you have downgraded to the correct version of the extension with
-   the `\dx` command. The output should show the downgraded version number.
+1. **Check that you have downgraded to the correct version TimescaleDB**
```
Comment (atovpeko) [inline self-hosted/upgrades/downgrade.md]:
```suggestion
1. **Check that you have downgraded to the correct version of TimescaleDB**
```

File: _partials/_plan_upgrade.md (hunk 2 above)
Comment (billy-the-fish) [reply _partials/_plan_upgrade.md]: Nice spot, fixed.

File: self-hosted/upgrades/minor-upgrade.md (hunk 5 above)
Comment (billy-the-fish) [reply self-hosted/upgrades/minor-upgrade.md]: Fixed.

File: self-hosted/upgrades/upgrade-pg.md (hunk 6 above)
Comment (billy-the-fish) [reply self-hosted/upgrades/upgrade-pg.md]: Fixed.

File: self-hosted/upgrades/major-upgrade.md [hunk 9]
```diff
@@ -1,141 +1,151 @@
...
-TimescaleDB&nbsp;2.5 to TimescaleDB&nbsp;2.6, see the
-[minor upgrades section][upgrade-minor].
+When you perform a major upgrade, new policies are automatically configured based on your current 
+configuration. In order to verify your policies post upgrade, in this upgrade process you export 
+your policy settings before upgrading.
 
 <ConsiderCloud />
 
-## Plan your upgrade
+This page shows you how to perform a major upgrade, for minor upgrades, see
```
Comment (atovpeko) [inline self-hosted/upgrades/major-upgrade.md]:
```suggestion
This page shows you how to perform a major upgrade. For minor upgrades, see
```

File: self-hosted/upgrades/major-upgrade.md [hunk 10]
```diff
@@ -1,141 +1,151 @@
...
+1. Upgrade TimescaleDB to 2.17.2.
+
+You may need to [upgrade to the latest PostgreSQL version][upgrade-pg] before you upgrade TimescaleDB.
+
+<SupportMatrix />
 
-When you upgrade from TimescaleDB&nbsp;1, to TimescaleDB&nbsp;2, scripts
+## Check for failed retention policies
+
+When you upgrade from TimescaleDB 1, to TimescaleDB 2, scripts
```
Comment (atovpeko) [inline self-hosted/upgrades/major-upgrade.md]:
```suggestion
When you upgrade from TimescaleDB 1 to TimescaleDB 2, scripts
```

File: self-hosted/upgrades/minor-upgrade.md [hunk 11]
```diff
@@ -1,66 +1,48 @@
...
+# Upgrade TimescaleDB to a new minor version
 
-A minor upgrade is when you upgrade within your current major version of
-TimescaleDB. For example, when you upgrade from TimescaleDB&nbsp;2.5, to
-TimescaleDB&nbsp;2.6.
-
-For upgrading to a new major version, for example upgrading from
-TimescaleDB&nbsp;1 to TimescaleDB&nbsp;2, see the
-[major upgrades section][upgrade-major].
+A minor upgrade is when you update from TimescaleDB `<major version>.x`, to TimescaleDB `<major version>.y`.
```
Comment (atovpeko) [inline self-hosted/upgrades/minor-upgrade.md]:
```suggestion
A minor upgrade is when you update from TimescaleDB `<major version>.x` to TimescaleDB `<major version>.y`. 
```

File: self-hosted/upgrades/upgrade-pg.md [hunk 12]
```diff
@@ -1,100 +1,88 @@
...
 
 # Upgrade PostgreSQL
 
-Because TimescaleDB is a PostgreSQL extension, you need to ensure you keep your
-underlying PostgreSQL installation up to date. When you upgrade your TimescaleDB
-extension to a new version, always take the time to check if you need to also
-upgrade your PostgreSQL version. If you are running an older version of
-PostgreSQL, you need to upgrade it first, before you upgrade your TimescaleDB
-extension.
+TimescaleDB is a PostgreSQL extension. When you upgrade, ensure that you are installing to compatible versions of
```
Comment (atovpeko) [inline self-hosted/upgrades/upgrade-pg.md]: the compatible versions?

File: self-hosted/upgrades/upgrade-pg.md [hunk 13]
```diff
@@ -1,100 +1,88 @@
...
-1.  At the psql prompt, perform the upgrade:
+   Set the `OLD_BIN_DIR` environment variable to the folder holding the `postgres` binary. 
+   For example, `which postgres` returns something like `/usr/lib/postgresql/16/bin/postgres`.
+   ```bash
+   export OLD_BIN_DIR=/usr/lib/postgresql/16/bin
+   ``` 
+
+1. **Set your connection string**
+
+   This variables hold the connection information for the database to upgrade:
```
Comment (atovpeko) [inline self-hosted/upgrades/upgrade-pg.md]:
```suggestion
   This variable holds the connection information for the database to upgrade:
```

**Review by atovpeko — CHANGES_REQUESTED** [review]
A few minor things

File: _partials/_plan_upgrade.md (hunk 7 above)
Comment (billy-the-fish) [reply _partials/_plan_upgrade.md]: Removed

File: self-hosted/upgrades/upgrade-pg.md (hunk 12 above)
Comment (billy-the-fish) [reply self-hosted/upgrades/upgrade-pg.md]: Updated.

**Review by atovpeko — APPROVED** [review]

**Pull request merged**